### PR TITLE
Feature1

### PR DESCRIPTION
**Pull request by Vlad-Andreychik**

### CODE REVIEW

File: fixtures/rabota_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+from clients.http_client import HTTPClient
+from models.parser_jobs_tut_by import RabotaByParser
+
+url = 'https://rabota.by/search/' \
+      'vacancy?L_is_autosearch=false&area=16&clusters=true&enable_snippets=true&text=python&page=0'
+HEADERS = {
+    'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/webkit-version (KHTML, like Gecko) '
+                  'Silk / browser - version like Chrome / chrome - version Safari / webkit - version'}
+
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py]: url and Headers should not here, put this vars into fixture which uses these vars

File: fixtures/rabota_fixtures.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+@pytest.fixture
+def get_response():
+    return HTTPClient.get(url, headers=HEADERS)
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py]: docstring

File: fixtures/rabota_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+
+@pytest.fixture
+def get_response_from_wrong_url():
+    wrong_url = 'https://rabota.by/search/vacncy?clusters=true&' \
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py:28]: docstring

File: fixtures/rabota_fixtures.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+@pytest.fixture
+def get_description_for_first_vacancy():
+    response = HTTPClient.get(url, headers=HEADERS).text
+    lxml_text = RabotaByParser.get_lxml(response)
+    list_id = RabotaByParser.get_list_of_vacancy_id(lxml_text)
+    first_vacancy_id = list_id[0]
+    response_for_first_vacancy = HTTPClient.get(f'https://rabota.by/vacancy/{first_vacancy_id}?query=python',
+                                                headers=HEADERS).text
+    description = RabotaByParser.find_vacancy_description(response_for_first_vacancy)
+    return description.text.lower()
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py]: lxml_text = RabotaByParser.get_lxml(response)
    list_id = RabotaByParser.get_list_of_vacancy_id(lxml_text)
    first_vacancy_id = list_id[0]
    response_for_first_vacancy = HTTPClient.get(f'https://rabota.by/vacancy/{first_vacancy_id}?query=python',
                                                headers=HEADERS).text
    description = RabotaByParser.find_vacancy_description(response_for_first_vacancy)
    return description.text.lower()


all this logic should be implemented inside Parser class

File: fixtures/rabota_fixtures.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        count_flask = RabotaByParser.count_word('flask', description)
+        dictionary[f'{number} vacancy'] = {'python': count_python, 'linux': count_linux, 'flask': count_flask}
+        python += count_python
+        linux += count_linux
+        flask += count_flask
+        number += 1
+    avg_python = python // len(all_urls_list)
+    avg_linux = linux // len(all_urls_list)
+    avg_flask = flask // len(all_urls_list)
+    return avg_python, avg_linux, avg_flask
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py]: Incorrect logic at all. Fixture should not contain such logic, stlit into several and put Parser logic into its method.
Fixture should not return average count of 3 words, it should return for 1.

File: fixtures/rabota_fixtures.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+
+@pytest.fixture
+def amount_of_vacancies_founded():
+    response = HTTPClient.get(url, headers=HEADERS).text
+    lxml_text = RabotaByParser.get_lxml(response)
+    h1_text = lxml_text.find('h1', class_='bloko-header-1').text
+    pattern = '\d+'
+    return int(re.findall(pattern, h1_text)[0])
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py]: lxml_text = RabotaByParser.get_lxml(response)
    h1_text = lxml_text.find('h1', class_='bloko-header-1').text
    pattern = '\d+'
    return int(re.findall(pattern, h1_text)[0])



Your Parser class should do it

File: fixtures/rabota_fixtures.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+
+@pytest.fixture
+def amount_vacancies_on_one_page(get_response):
+    lxml_text = RabotaByParser.get_lxml(get_response.text)
+    list_id = RabotaByParser.get_list_of_vacancy_id(lxml_text)
+    return len(list_id)
```
Comment (akopylov89) [inline fixtures/rabota_fixtures.py]: lxml_text = RabotaByParser.get_lxml(response)
    h1_text = lxml_text.find('h1', class_='bloko-header-1').text
    pattern = '\d+'
    return int(re.findall(pattern, h1_text)[0])


Your Parser class should do it

File: tests/rabota_by_tests.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    test connection of the web site
+    """
+    assert get_response.status_code == 200
+
+
+def test_shotgun_not_found(get_query_shotgun):
+    """
+    test that there are no search results for the word "shotgun"
+    """
+    assert 'По запросу «shotgun» ничего не найдено' in get_query_shotgun.text
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: no need to write docstrings for tests, remove

File: tests/rabota_by_tests.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    actual_number_of_pages = RabotaByParser.amount_of_pages(get_response.text)
+    estimated_number_of_pages = math.ceil(amount_of_vacancies_founded / amount_vacancies_on_one_page)
+    assert estimated_number_of_pages == int(actual_number_of_pages)
+
+
+def test_wrong_url(get_response_from_wrong_url):
+    """
+    test connection with wrong url
+    """
+    assert get_response_from_wrong_url.status_code == 404
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: assert message

File: tests/rabota_by_tests.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def test_accordance_amount_pages_for_amount_of_vacancies(amount_of_vacancies_founded, amount_vacancies_on_one_page,
+                                                         get_response):
+    """
+    test that there is correct number of pages for our amount of vacancies
+    """
+    actual_number_of_pages = RabotaByParser.amount_of_pages(get_response.text)
+    estimated_number_of_pages = math.ceil(amount_of_vacancies_founded / amount_vacancies_on_one_page)
+    assert estimated_number_of_pages == int(actual_number_of_pages)
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: assert message

File: tests/rabota_by_tests.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    assert count_python in range(get_avg_occurrences_for_words[0] - 1, get_avg_occurrences_for_words[0] + 2)
+    assert count_linux in range(get_avg_occurrences_for_words[1] - 1, get_avg_occurrences_for_words[1] + 2)
+    assert count_flask in range(get_avg_occurrences_for_words[2] - 1, get_avg_occurrences_for_words[2] + 2)
+
+
+def test_amount_vacancies_on_one_page(amount_vacancies_on_one_page):
+    """
+    test that amount vacancies on one page is equal to 50
+    """
+    assert amount_vacancies_on_one_page == 50
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: assert message

File: tests/rabota_by_tests.py [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    test that the occurrence of words "python", "linux", "flask" is within boundaries avg +- 1
+    """
+    description = get_description_for_first_vacancy
+    count_python = RabotaByParser.count_word('python', description)
+    count_linux = RabotaByParser.count_word('linux', description)
+    count_flask = RabotaByParser.count_word('flask', description)
+    assert count_python in range(get_avg_occurrences_for_words[0] - 1, get_avg_occurrences_for_words[0] + 2)
+    assert count_linux in range(get_avg_occurrences_for_words[1] - 1, get_avg_occurrences_for_words[1] + 2)
+    assert count_flask in range(get_avg_occurrences_for_words[2] - 1, get_avg_occurrences_for_words[2] + 2)
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: assert message

File: tests/rabota_by_tests.py [hunk 13]
```diff
@@ -0,0 +1,54 @@
+import math
+
+from models.parser_jobs_tut_by import RabotaByParser
+
+
+def test_connection(get_response):
+    """
+    test connection of the web site
+    """
+    assert get_response.status_code == 200
```
Comment (akopylov89) [inline tests/rabota_by_tests.py]: assert message

File: fixtures/rabota_fixtures.py (hunk 1 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Done

File: fixtures/rabota_fixtures.py (hunk 2 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Done

File: fixtures/rabota_fixtures.py (hunk 3 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py:28]: Done

File: fixtures/rabota_fixtures.py (hunk 4 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Done

File: fixtures/rabota_fixtures.py (hunk 5 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Still in progress

File: fixtures/rabota_fixtures.py (hunk 6 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Done

File: fixtures/rabota_fixtures.py (hunk 7 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Done

File: tests/rabota_by_tests.py (hunk 8 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Done

File: tests/rabota_by_tests.py (hunk 9 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Done

File: tests/rabota_by_tests.py (hunk 10 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Done

File: tests/rabota_by_tests.py (hunk 11 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Done

File: tests/rabota_by_tests.py (hunk 12 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Done

File: tests/rabota_by_tests.py (hunk 13 above)
Comment (Vlad-Andreychik) [reply tests/rabota_by_tests.py]: Done

File: fixtures/rabota_fixtures.py (hunk 5 above)
Comment (Vlad-Andreychik) [reply fixtures/rabota_fixtures.py]: Done

**Pull request closed without merging**